packages/flutter/lib/src/cupertino/tab_view.dart

```diff
@@ -162,15 +162,39 @@ class _CupertinoTabViewState extends State<CupertinoTabView> {
           ..add(_heroController);
   }
 
+  GlobalKey<NavigatorState>? _ownedNavigatorKey;
+  GlobalKey<NavigatorState> get _navigatorKey {
+    if (widget.navigatorKey != null) {
+      return widget.navigatorKey!;
+    }
+    _ownedNavigatorKey ??= GlobalKey<NavigatorState>();
+    return _ownedNavigatorKey!;
+  }
+
+  // Whether this tab is currently the active tab.
+  bool get _isActive => TickerMode.of(context);
+
   @override
   Widget build(BuildContext context) {
-    return Navigator(
-      key: widget.navigatorKey,
+    final Widget child = Navigator(
+      key: _navigatorKey,
       onGenerateRoute: _onGenerateRoute,
       onUnknownRoute: _onUnknownRoute,
       observers: _navigatorObservers,
       restorationScopeId: widget.restorationScopeId,
     );
+
+    // Handle system back gestures only if the tab is currently active.
+    return NavigatorPopHandler(
+      canPop: _isActive ? null : true,
```

Comment:
If they really want the use case of canPop: false, they should just use `PopScope`

Reply:
You're right, it should be enable/disable.  The reason why I added it was to support CupertinoTabScaffold ([code](https://github.com/flutter/flutter/commit/814e202be23060fd137abcd51052a50b2270e531#diff-478b7dfa524d168e545b2a81ad177b993691938f7f7132f11122e95b3ef7ac2bR188)).  I needed to enable/disable the PopScope without removing it from the widget tree, because if I remove it then it won't receive notifications.  I'll change it to `enabled`.